src/Watchers/Warden.Watchers.ServerStatus/ITcpClient.cs

```diff
@@ -0,0 +1,60 @@
+﻿namespace Warden.Watchers.ServerStatus
+{
+    using System;
+    using System.Net;
+    using System.Net.Sockets;
+    using System.Threading.Tasks;
+
+    public interface ITcpClient : IDisposable
+    {
+        /// <summary>
+        /// Connects to specified ip address via TCP protocol
+        /// </summary>
+        /// <param name="ip">A ip address of server/</param>
+        /// <param name="port">A port name of server to connect.</param>
+        /// <returns>Task that perfroms a connection.</returns>
+        Task ConnectAsync(IPAddress ip, int port, TimeSpan? timeout);
+
+        /// <summary>
+        /// Gets a flag that indicates wheter current instance is connected to any server.
+        /// </summary>
+        bool IsConnected { get; }
+    }
+
+    internal class DefaultTcpClient : ITcpClient
```

Comment:
Yea, right - in all my projects I have used "Default" prefix to indicate that it is default implementation used. Just convention. 
Changed to *TcpClient*.